integrations/nifi/nifi-nar-bundles/nifi-hadoop-bundle/nifi-hadoop-processors/src/main/java/com/thinkbiganalytics/nifi/v2/hdfs/CreateHDFSFolder.java

```diff
@@ -177,7 +177,11 @@ public void onScheduled(ProcessContext context) throws IOException {
         final short dfsUmask = resolveUMask(umaskProp);
 
         final Configuration conf = getConfiguration();
-        FsPermission.setUMask(conf, new FsPermission(dfsUmask));
+        short oldUmask = Short.parseShort(conf.get(FsPermission.UMASK_LABEL));
+        if (oldUmask != dfsUmask) {
+            FsPermission.setUMask(conf, new FsPermission(dfsUmask));
+            resetFileSystem();
+        }
     }
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.apache.nifi.processor.exception.ProcessException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)